app/src/infra/attributes.rs

```diff
@@ -0,0 +1,150 @@
+pub struct AttributeDescription {
+    pub attribute_identifier: String,
+    pub attribute_name: String,
+    pub aliases: Vec<String>,
+}
+
+pub mod group {
+
+    use super::AttributeDescription;
+
+    pub fn resolve_group_attribute_description(name: &str) -> Option<AttributeDescription> {
```

Comment:
I think we can have a single function for both user and group attributes together, there's no need for the split. And ideally also refactor the display code above for code sharing.

Reply:
I kind of think that might be asking for trouble, displayName doesn't have the same aliases for users and groups, and one of those aliases is "id", which means something else for users. Personally, I think it'd be best to just keep them separate.

They still use the same struct, so I can easily make some shared display code anyway. 

By the way, I didn't do that, because I didn't see a good place for it, and thought it small enough to not matter too much. Any suggestions of where? Maybe a "fragments" folder under "components"?

Reply:
Pushed an update with shared display code (will probably update that a bit when revisiting attribute validation PR). Added a "fragments" folder for the shared display code.

For now, the user and group attributes are still split.

Reply:
That makes sense, I didn't consider "displayName". If the names had been non-overlapping I would have pushed for it (YAGNI) but in this case you're right.

I guess "fragment" is as good a name as any, although it stinks of "utils". I don't have a better name.

Reply:
Agree with the "utils"-stench. Should at least be fairly simple to rename, if we find some inspiration at some point.

I'll mark this as resolved then.